source/include/core_sntp_client.h

```diff
@@ -0,0 +1,463 @@
+/*
+ * coreSNTP v1.0.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_sntp_client.h
+ * @brief API of an SNTPv4 client library that can send time requests and receive time response to/from
+ * SNTP/NTP servers. The library follows the Best Practices suggested in the the SNTPv4 specification,
+ * [RFC 4330](https://tools.ietf.org/html/rfc4330).
+ * The library can be used to run an SNTP client in a dedicated deamon task to periodically synchronize
+ * time from the Internet.
+ */
+
+#ifndef CORE_SNTP_CLIENT_H_
+#define CORE_SNTP_CLIENT_H_
+
+/* Standard include. */
+#include <stdint.h>
+#include <stdbool.h>
```

Comment:
This is interesting but some compilers actually need `#include <stdlib.h>` for `size_t`. Otherwise, this won't compile. We found this out while setting up CBMC. Turns out, `goto-cc` requires it.

Reply:
Good catch about including header for `size_t` definition. However, the [ISO C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) mentions (in Section 7.17) that the type is defined in `stddef.h`. I will include this header instead